File: from-3101-to-3200/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.go
func longestMonotonicSubarray(nums []int) int {
    incr, decr := 1, 1
    incrMax, decrMax := 1, 1
    prev := nums[0]

    for _, num := range nums[1:] {
        if num > prev {
            if decr > decrMax {
                decrMax = decr
            }
            incr += 1
            decr = 1
        } else if num < prev {
            if incr > incrMax {
                incrMax = incr
            }
            incr = 1
            decr += 1
        } else {
            if incr > incrMax {
                incrMax = incr
            }
            if decr > decrMax {
                decrMax = decr
            }
            incr, decr = 1, 1
        }
        prev = num
    }
    if incr > incrMax {
        incrMax = incr
    }
    if decr > decrMax {
        decrMax = decr
    }
    if incrMax > decrMax {
        return incrMax
    }
    return decrMax
}

File: from-3101-to-3200/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/solution.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        incr, decr = 1, 1
        prev = nums[0]
        incr_max, decr_max = 0, 0
        for num in nums[1:]:
            if num > prev:
                decr_max = max(decr_max, decr)
                incr += 1
                decr = 1
            elif num < prev:
                incr_max = max(incr_max, incr)
                incr = 1
                decr += 1
            else:
                incr_max = max(incr_max, incr)
                decr_max = max(decr_max, decr)
                incr, decr = 1, 1

            prev = num
            
        return max(max(incr_max, incr), max(decr_max, decr))